R/pool-methods.R

```diff
@@ -0,0 +1,89 @@
+#' @include pool.R
+NULL
+
+#' S4 class for compatibility with DBI methods
+#' @export
+setClass("Pool")
+
+## documented manually, with the Pool object
+#' @export
+poolCreate <- function(factory, ...,
+                       minSize = 1, maxSize = Inf,
+                       idleTimeout = 60000, validateTimeout = 2000,
+                       stateEnv = NULL) {
+  pool <- Pool$new(
+    function() {factory(...)},
+    minSize, maxSize,
+    idleTimeout, validateTimeout, stateEnv
+  )
+  ## if a createPool is used with a DBIDriver, make a note
+  ## (to ease dplyr compatibility later on)
+  checkDriver(pool, ...)
```

Comment:
1) Yes, just because that's the sort of thing `stateEnv` is for. No big deal I guess.

2) I really hate that idea, sorry. `dbPool` is a DBI pool that features compatibility with `dplyr`, `poolCreate` is a generic pool-creating function that we won't even really tell normal users about--if you use the wrong function, you get the wrong results.